R/zzz.r

```diff
@@ -49,5 +49,12 @@ NULL
 
   find_rtools()
 
+  shims <<- list(
```

Comment:
It looks like the namespace is not locked until after `.onLoad()` is
called, so it does work with a regular load as well.  I get no errors after
installing the package in this pull request and (in a new session) calling
`library(devtools)`.

My initial pull request used an environment as you describe and populated
the environment when `load_all` was called for the first time, but this
method seems simpler as there is no need for these objects to change after
the package is loaded.

C code missing was a silly mistake, thank you for pointing it out.

The situation this pull request is trying to remedy is when one package
("test_package") is loaded with `load_all()`.  And a second package calls
one of the shim'd functions, e.g. `system.file("unitTests", package =
"test_package")`.  Would what you proposed work in that case?